src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -317,10 +322,10 @@ export class TerminalService implements ITerminalService {
 		return terminalProcess;
 	}
 
-	public static createTerminalEnv(parentEnv: IStringDictionary<string>, shell: IShell, workspace: IWorkspace, locale?: string): IStringDictionary<string> {
+	public static createTerminalEnv(parentEnv: IStringDictionary<string>, shell: IShell, workspace: IWorkspace, shellPath?: string, locale?: string): IStringDictionary<string> {
```

Comment:
Thinking about this again, I think we should probably support args for the shellPath too (what `IShell` does). Meaning shellPath and shellArgs should pass through the API layers.